### DISCUSSION TITLE
[BUG]: receiving not existent orders from getShopReceipts endpoint

### DISCUSSION THREAD

**Discussion by KushGoyal** [etsy/open-api#706] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am using `getShopReceipts` to fetch orders. One of the order I received from the endpoint has id `2651714947` this order does not exist when trying to fetch the order using `getShopReceipt`. My code did not generate the order by itself, this was for sure received from the `getShopReceipts` endpoint. The customer is complaining about fantom orders being received from the etsy API because this has happened before too.
Can you please explain why are such orders being sent in the response.

### Steps to reproduce

It is not possible to reproduce this again because the `getShopReceipts` sends non existent orders irregularly.

### Expected behavior

Only orders which can be found in the etsy shop admin dashboard should be sent.

### Additional context

I have tried using `is_paid=True` query param but that does not solve the problem because a lot orders get missed when using this.

**Comment by etsyachristensen** [comment]
@KushGoyal Can you provide the full set of results that included that receipt id? I don't see that ID in our system anywhere. What shop id is having this issue? we will need to be able to recreate the problem in order to find a resolution.

**Comment by KushGoyal** [comment]
Hi @etsyachristensen I do not have the response which included this order id. But I can share the shop id, receipt id and transaction id included in the order.

shop id `24973314`
receipt id `2651714947`
transaction id `3250161924 `

**Comment by etsyachristensen** [comment]
@KushGoyal We can't find any record of the receipt or transaction ids anywhere in our system. They don't belong to someone else, nor are they somehow deleted. They just never existed. We haven't been able to duplicate your experience. Is there any way you can capture the full response the next time this happens and email the full response and note which ids are bad? You can email it to developers@etsy.com and reference this github issue and ask that they forward your email to me?

**Comment by KushGoyal** [comment]
Hi @etsyachristensen I will need to modify my code to log the response every time I make a request. Because I have no way of knowing when this will happen. I am not sure if that is possible. However I can confirm you that my app is not making up the ids. Those were received from the etsy api response. If I am able to find a way to capture the response I will surely share it with you by email.

**Comment by etsyachristensen** [comment]
I don't doubt your sincerity @KushGoyal . I trust that it's showing up that way but it's about impossible for us to troubleshoot without more data. If we can get a full output, along with the query params and/or headers it will help us to dig that up. If you can also log the response `etsy-request-uuid` header that might also help as well.

**Comment by etsyachristensen** [comment]
@KushGoyal Do you have any recent logs where this is happening? I'm going to convert this issue to a discussion as we will be shutting down the Issues board. We can continue to track this in Discussions.

**Reply by KushGoyal** [reply]
I have not been able to find a way to log this. I use the `getShopReceipts` endpoint to fetch the orders. But to verify if the fetched orders exist on etsy I will have to do a get request for each order id received. This is because users are usually not checking this manually. It is by chance that someone finds an order which does not exist.

**Comment by FloppyChips** [comment]
I have just created https://github.com/etsy/open-api/discussions/880 but I wonder if it is the same issue.  I have asked our customer who owns the Etsy shop to confirm if the duplicates we see from _getShopReceipts_ are still present in the backend Shop Manager.
